packages/reputation-miner/ReputationMinerClient.js

```diff
@@ -225,24 +227,46 @@ class ReputationMinerClient {
 
       this.lockedForBlockProcessing = true;
       // DO NOT PUT ANY AWAITS ABOVE THIS LINE OR YOU WILL GET RACE CONDITIONS
+      // When you leave this function, make sure to call this.endDoBlockChecks() to unlock
 
       if (this.blockTimeoutCheck) {
         clearTimeout(this.blockTimeoutCheck);
       }
 
       const block = await this._miner.realProvider.getBlock(blockNumber);
       const addr = await this._miner.colonyNetwork.getReputationMiningCycle(true);
-
       if (addr !== this.miningCycleAddress) {
-        // Then the cycle has completed since we last checked. Let's sort out our potential submissions for the next cycle.
-        await this.processReputationLog();
-        this.best12Submissions = await this.getTwelveBestSubmissions();
-        this.miningCycleAddress = addr;
+        // Then the cycle has completed since we last checked.
         if (this.confirmTimeoutCheck) {
           clearTimeout(this.confirmTimeoutCheck);
         }
-        // If we don't see this cycle completed in the next day and ten minutes, then report it
+        // If we don't see this next cycle completed in the next day and ten minutes, then report it
         this.confirmTimeoutCheck = setTimeout(this.reportConfirmTimeout.bind(this), (24 * 3600 + 600) * 1000);
+
+        // Let's process the reputation log if it's been ten blocks
+        if (this.blocksSinceCycleCompleted < 10) {
```

Comment:
Remind me why we process the reputation log every ten blocks? If it's the active mining cycle, it's no longer being updated, so why keep checking? And if it's the inactive one, why process it before it's active?

Reply:
So if this line is running, a mining cycle has completed, and we've not processed the next log yet. My intention here is to wait until it has been ten blocks to process the next log to avoid falling foul of a block reorganisation that doesn't include the transaction that completes the cycle; if that happens, then the log processing will fail as after the reorganisation, it will be attempting to query a contract that doesn't exist.

Once the log has been processed, it doesn't get processed again - it is certainly not meant to be running every ten blocks (and if it is, that's very wrong, as you say!)

This is my best guess as to the intermittent error we were seeing on goerli, which did coincide with reorgs each time it was seen.

Reply:
Ah I see, because the miner isn't set up to correct itself for a reorg, you're basically waiting for 10 blocks as a type of probabilistic finality.